Systemize fixed point types to be tested

kimikage opened the pull request:
cf. https://github.com/JuliaMath/FixedPointNumbers.jl/issues/139#issuecomment-664006515

This does not commonize the tests between `Fixed` and `Normed`, but does prepare for the commonization.

timholy commented:
Due to code movement this won't be easy to review, and it doesn't look like it's much in need anyway. So just go ahead and merge when you're ready.

kimikage commented:
Thank you for your response. Most of the code movement is not important, but I don't want to mess with the arrangement every time I add a test set. :sweat_smile:

The essential change in this PR is https://github.com/JuliaMath/FixedPointNumbers.jl/pull/208/commits/2a4a61ba207d6f75e83b231e41a22e789364a3d9.
TBH, I don't really like the new `target` function because I think the test code should be simple. However, test case "justification" is important, and this change saves more than 15 seconds of testing time on my local machine.

timholy commented:
Test performance is definitely important. Maybe add a little bit of commentary about the heavy/default/thin/light system? Just from reading it's not entirely obvious since the `target_f_series` is elsewhere.

kimikage commented:
I moved the `target_f_series` to "test/common.jl".  (it is contrary to the philosophy of  specialization, though. :smile:)
Also, I added the comment about the `heavy`/`default`/`light`/`thin` system as the docstring for `target_f`.

timholy commented:
LGTM!
